core/state/pruner/pruner.go

```diff
@@ -370,6 +371,11 @@ func dumpRawTrieDescendants(db ethdb.Database, root common.Hash, output *stateBl
 	for i := 0; i < threads; i++ {
 		results <- nil
 	}
+	// We also create a semaphore for the storage trie traversal, to limit the number of goroutines
+	resultsPerAccount := make(chan error, threads)
```

Comment:
Why not just use `results` instead of creating a new semaphore? As-is you'll likely end up with 2*threads running, as you have one set of threads for the top level and one set of threads for account storage.

Reply:
Yeah i am kind of stuck on this and can use your help, so what is happening is earlier there was only one level of multithreading across account, and now its a multilevel threading across accounts and across hashes.

So whats happening is lets say there there are 10 account and 10 threads allowed, 

All the 10 threads are consumed by 10 accounts, and when the time comes to open the a new thread for pruning across hashes there are no more threads left and its a deadlock (earlier this was not an issue since the same thread that was used for the account could be used for pruning the hashes.)

Wanted to get your thoughts on how to handle this case.

Reply:
Why not move the `storageTr.NodeIterator(big.NewInt((i - 1) << 3).Bytes())` call into the inner storage goroutine, and then drop the outer goroutine and only have that happen in the `if !config.ParallelPrune` case? Right now if ParallelPrune is enabled, no significant computation is happening in the accounts goroutine, and this'll collapse it down to always only being one level of goroutines.

Reply:
Done.